realm/realm-library/src/androidTest/java/io/realm/CustomRealmNameTests.java

```diff
@@ -0,0 +1,220 @@
+        RealmObjectSchema classSchema = realm.getSchema().get(ClassWithPolicy.CLASS_NAME);
+        for (String field : ClassWithPolicy.ALL_FIELDS) {
+            assertTrue(field + " was not found.", classSchema.hasField(field));
+        }
+    }
+
+    // Check that explicit class name override both module and class policies
+    @Test
+    public void className_overrideModuleClassPolicy() {
+        assertTrue(realm.getSchema().contains(ClassNameOverrideModulePolicy.CLASS_NAME));
+    }
+
+    // Check that a explicitly setting a field name overrides a class field name policy
+    @Test
+    public void fieldName_overrideClassPolicy() {
+        RealmObjectSchema classSchema = realm.getSchema().get(FieldNameOverrideClassPolicy.CLASS_NAME);
+        assertTrue(classSchema.hasField(FieldNameOverrideClassPolicy.FIELD_CAMEL_CASE));
+    }
+
+    // Check that a explicitly setting a field name overrides a module field name policy
+    @Test
+    public void fieldName_overrideModulePolicy() {
+        RealmObjectSchema classSchema = realm.getSchema().get(FieldNameOverrideClassPolicy.CLASS_NAME);
+        assertTrue(classSchema.hasField(FieldNameOverrideClassPolicy.FIELD_CAMEL_CASE));
+    }
+
+    //
+    // Query tests
+    //
+    // Mostly smoke test, as we only want to test that the query system correctly maps between
+    // Java field names and cores.
+    //
+    @Test
+    public void typedQueryWithJavaNames() {
+        RealmResults<ClassWithPolicy> results = realm.where(ClassWithPolicy.class)
+                .equalTo("camelCase", "foo") // Java name in model class
+                .equalTo("parents.PascalCase", 1) // Backlinks also uses java names
+                .sort("mHungarian") // Sorting uses Java names
+                .distinctValues("customName") // Distinct uses Java names
+                .findAll();
+        assertTrue(results.isEmpty());
+    }
+
+    @Test
+    public void typedQueryWithInternalNamesThrows() {
+
+        // Normal predicates
+        try {
+            realm.where(ClassWithPolicy.class).equalTo(ClassWithPolicy.FIELD_CAMEL_CASE, "");
+        } catch (IllegalArgumentException ignore) {
+        }
+
+        // Sorting
+        try {
+            realm.where(ClassWithPolicy.class).sort(ClassWithPolicy.FIELD_CAMEL_CASE);
+        } catch (IllegalArgumentException ignore) {
+        }
+
+        // Distinct
+        try {
+            realm.where(ClassWithPolicy.class).distinctValues(ClassWithPolicy.FIELD_CAMEL_CASE);
+        } catch (IllegalArgumentException ignore) {
+        }
+
+        // Backlinks do not exist as internal fields that can be queried
+    }
+
+
+    @Test
+    public void dynamicQueryWithInternalNames() {
+        // Backlink queries not supported on dynamic queries
+        RealmResults<DynamicRealmObject> results = dynamicRealm.where(ClassWithPolicy.CLASS_NAME)
+                .equalTo(ClassWithPolicy.FIELD_CAMEL_CASE, "foo") // Java name in model class
```

Comment:
Java name? shouldn't be internal name for Dynamic Realm. same for below

Reply:
If the class no longer exists, then you can't obtain any field-specific / class-specific policy configuration for it, though.

Reply:
DynamicRealms has to work with the assumption that no java model classes exists. That this test case actually references `ClassWithPolicy` is just so it is easier to maintain the test case.

But you are correct, DynamicRealms do not understand the concept of name policies as that concept isn't relevant to them.